### PR TITLE
LED 2 color pattern code

### PR DESCRIPTION
**Pull request by WillHamel8727**
#6

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Indications.java [hunk 1]
```diff
@@ -1,33 +1,82 @@
...
+   m_LedStrip = new AddressableLED(kSensors.ledPort);
+   m_LedData = new AddressableLEDBuffer(kSensors.ledLength);
+   m_LedStrip.setData(m_LedData);
+   m_LedStrip.start();
+ }
+
+
+ public void pattern2(RGB color1,RGB color2, float colorSpeed){
+   // 2 Colored Pattern
+    var int patternSize = 2;
```
Comment (dasaldino1) [inline src/main/java/frc/robot/subsystems/Indications.java:51]:
```suggestion
   int patternSize = 2;

```

File: src/main/java/frc/robot/subsystems/Indications.java [hunk 2]
```diff
@@ -1,33 +1,82 @@
...
+   for (var i = 1; i < m_ledData.getLength(); i++) {
```
Comment (lance) [inline src/main/java/frc/robot/subsystems/Indications.java:52]: You also need to declare the `offset` variable that is used below in the `for()` loop and after. Do that just before the loop, like this.
```suggestion
    int offset = 0;
    for (var i = 1; i < m_ledData.getLength(); i++) {
```

File: src/main/java/frc/robot/subsystems/Indications.java [hunk 3]
```diff
@@ -1,33 +1,82 @@
...
+    //color 1
+     if ((i + offset) % patternSize) < 1 {
+     m_ledData.setRGB(i, color1.red, color1.green, color1.blue);
```
Comment (lance) [inline src/main/java/frc/robot/subsystems/Indications.java:55]: Capitalized `L` in `m_LedData`.
```suggestion
     m_LedData.setRGB(i, color1.red, color1.green, color1.blue);

```

File: src/main/java/frc/robot/subsystems/Indications.java [hunk 4]
```diff
@@ -1,33 +1,82 @@
...
+   m_LedStrip.start();
+ }
+
+
+ public void pattern2(RGB color1,RGB color2, float colorSpeed){
+   // 2 Colored Pattern
+    var int patternSize = 2;
+   for (var i = 1; i < m_ledData.getLength(); i++) {
+    //color 1
+     if ((i + offset) % patternSize) < 1 {
```
Comment (lance) [inline src/main/java/frc/robot/subsystems/Indications.java:54]: Fix parentheses - you need it around the full expression being evaluated in the `if` statement.
```suggestion
     if (((i + offset) % patternSize) < 1) {
```

File: src/main/java/frc/robot/subsystems/Indications.java [hunk 5]
```diff
@@ -1,33 +1,82 @@
...
+     m_ledData.setRGB(i, color1.red, color1.green, color1.blue);
+     } else {
+     //color 2
+     m_ledData.setRGB(i, color2.red, color2.green, color2.blue);
```
Comment (lance) [inline src/main/java/frc/robot/subsystems/Indications.java:58]: Capital `L` in `m_LedData` member variable.
```suggestion
     m_LedData.setRGB(i, color2.red, color2.green, color2.blue);

```

**Review by lance — APPROVED** [review]

**Pull request merged**